### PR TITLE
NAS-110081 / 22.02-RC.2 / Sentry integration - Errors should be logged at sentry.ixsystems.com automatically

### PR DESCRIPTION
**Pull request by RehanY147**
NAS-110081

### CODE REVIEW

File: src/app/pages/storage/volumes/volumes-list/volumes-list.component.ts [hunk 1]
```diff
@@ -192,7 +192,7 @@ export class VolumesListComponent extends EntityTableComponent implements OnInit
 
     this.has_key_dataset = {};
     this.has_encrypted_root = {};
-    this.ws.call('pool.dataset.query_encrypted_roots_keys').pipe(untilDestroyed(this)).subscribe((res) => {
+    this.ws.call('pool.dataset.query_encrypted_roots_keys').pipe(untilDestroyed(this)).subscribe((res: any) => {
```
Comment (undsoft) [inline src/app/pages/storage/volumes/volumes-list/volumes-list.component.ts:195]: Rogue `any`.

File: src/app/app.module.ts [hunk 2]
```diff
@@ -124,6 +127,22 @@ import { WebSocketService } from './services/ws.service';
...
+      }),
+    },
+    {
+      provide: Sentry.TraceService,
+      deps: [Router],
+    },
+    {
+      provide: APP_INITIALIZER,
+      useFactory: () => () => {},
+      deps: [Sentry.TraceService],
```
Comment (undsoft) [inline src/app/app.module.ts:143]: To check if initializer is actually needed.

File: src/app/views/sessions/signin/signin.component.ts [hunk 3]
```diff
@@ -382,6 +384,16 @@ export class SigninComponent implements OnInit, OnDestroy, AfterViewInit {
 
   successLogin(): void {
     this.snackBar.dismiss();
+
+    if (!environment.production) {
```
Comment (undsoft) [inline src/app/views/sessions/signin/signin.component.ts:388]: Best to find a way to enable this for nightly builds only.
If too hard, just switch the statement here.

**Review by undsoft — CHANGES_REQUESTED** [review]
We also need to honour the `Crash Reporting` setting in GUI Settings.

**Comment by RehanY147** [comment]
Creating a new PR to avoid merge conflicts

**Pull request closed without merging**